### PR TITLE
RUMM-3465: Update documentation for SDK v2

### PR DESCRIPTION
**Pull request by 0xnm**
### What does this PR do?

This PR updates documentation in the repository to be aligned with actual changes made for SDK v2.

Documentation in https://github.com/DataDog/documentation repository will be update in another PR.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-ndk/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Datadog Android Native Crash Collection
```
Comment (plousada) [inline features/dd-sdk-android-ndk/README.md:1]: We should also move this to the docs repo.

Comment (0xnm) [reply features/dd-sdk-android-ndk/README.md:1]: Fair point, I created a task for that.

**Review by plousada — APPROVED** [review]

File: dd-sdk-android-core/README.md [hunk 2]
```diff
@@ -22,129 +23,46 @@ class SampleApplication : Application() {
...
+See the dedicated [Datadog Android RUM Collection documentation][1] to learn how to send RUM data from your Android or Android TV application to Datadog.
 
-```kotlin
-    // This will add a tag "build_type:debug" or "build_type:release" accordingly
-    logger.addTag("build_type", BuildConfig.BUILD_TYPE)
-    
-    // This will add a tag "android"
-    logger.addTag("android")
-```
+## Setting up Datadog Logs SDK
```
Comment (rtrieu) [inline dd-sdk-android-core/README.md:57]:
```suggestion
## Setting up the Datadog Logs SDK
```

File: dd-sdk-android-core/README.md [hunk 3]
```diff
@@ -22,129 +23,46 @@ class SampleApplication : Application() {
...
-            .setNetworkInfoEnabled(true)
-            .setServiceName("com.example.app.android")
-            .setLogcatLogsEnabled(true)
-            .setDatadogLogsEnabled(true)
-            .setLoggerName("name")
-            .build()
-```
-
-### Logging
+### Using a Secondary Instance of the SDK
```
Comment (rtrieu) [inline dd-sdk-android-core/README.md:37]:
```suggestion
### Using a secondary instance of the SDK
```

File: dd-sdk-android-core/README.md [hunk 4]
```diff
@@ -1,13 +1,14 @@
-# Datadog SDK for Android
+# Datadog SDK for Android - core library
 
-## Getting Started 
+## Getting Started
```
Comment (rtrieu) [inline dd-sdk-android-core/README.md:3]:
```suggestion
## Getting started
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Some suggestions for consistency, but otherwise looks good!

**Pull request merged**